src/mbgl/renderer/layers/render_heatmap_layer.cpp

```diff
@@ -67,7 +67,7 @@ void RenderHeatmapLayer::render(PaintParameters& parameters, RenderSource*) {
                 }
             }
 
-            if (!renderTexture) {
+            if (!parameters.context.supportsHalfFloatTextures || !renderTexture) {
```

Comment:
Is the `!renderTexture` condition redundant in this case? If it supports half floats, the texture will be set earlier.

Reply:
There's that code that tries to bind the half-float texture and if an exception gets thrown resets the texture (in which case we'd have to recreate the texture here). I don't know the history of that, but it seems like intentional behavior...